app/uk/gov/hmrc/tai/service/IncomeService.scala

```diff
@@ -52,37 +53,70 @@ class IncomeService @Inject()(employmentService: EmploymentService,
 
   def updateTaxCodeIncome(nino: Nino, year: TaxYear, employmentId: Int, amount: Int)
                          (implicit hc: HeaderCarrier): Future[IncomeUpdateResponse] = {
-        for {
-          root <- taxAccountService.personDetails(nino)
-          cyResponse <- taxAccountRepository.updateTaxCodeAmount(nino, year, root.version, employmentId, NewEstimatedPay.code, amount)
-          incomeUpdateResponse <- updateCyPlusOneIfCySuccess(nino, year, employmentId, amount, root, cyResponse)
-        } yield {
-          auditor.sendDataEvent(
-            transactionName = "Update Multiple Employments Data",
-            detail = Map(
-              "nino" -> nino.value,
-              "year" -> year.toString,
-              "employmentId" -> employmentId.toString,
-              "newAmount" -> amount.toString))
-
-          incomeUpdateResponse
-        }
+
+    type IncomeUpdateForYear = (TaxYear, Int) => Future[HodUpdateResponse]
+    val taxCodeAmountUpdater: IncomeUpdateForYear = taxAccountRepository.updateTaxCodeAmount(nino, _, _, employmentId, NewEstimatedPay.code, amount)
+
+    val auditEventForIncomeUpdate: TaxYear => Unit = (taxYear: TaxYear) => {
+      auditor.sendDataEvent(
+        transactionName = "Update Multiple Employments Data",
+        detail = Map("nino" -> nino.value,
+                     "year" -> taxYear.toString,
+                     "employmentId" -> employmentId.toString,
+                     "newAmount" -> amount.toString))
+    }
+
+
+    taxAccountService.personDetails(nino) flatMap { root =>
+      if (year == TaxYear().next) {
+        updateTaxCodeIncomeCYPlusOne(nino, year, root, taxCodeAmountUpdater, auditEventForIncomeUpdate)
+      } else {
+        updateTaxCodeIncomeCY(nino, year, root, taxCodeAmountUpdater, auditEventForIncomeUpdate)
+      }
```

Comment:
Are we just going to silently fail if a invalid year is sent?

Reply:
Currently, that's what it does. I'll raise this behaviour and we can clarify what the behaviour should be